### PR TITLE
Add comments to linear_util.py.

### PR DESCRIPTION
**Pull request by hawkinsp**

### CODE REVIEW

File: jax/linear_util.py [hunk 1]
```diff
@@ -12,6 +12,59 @@
...
+
+For example:
+
+from jax import linear_util as lu
+
+# A transformation that scales its argument down and its result up.
+@lu.transformation
+def scale_transformer(scale, x):
+  ans = yield (x / scale,)
+  yield (x * scale,)
```
Comment (mattjj) [inline jax/linear_util.py]: This doesn't have to be a tuple actually. It's for the usual reason of functions having multiple arguments but single outputs.

File: jax/linear_util.py [hunk 2]
```diff
@@ -12,6 +12,59 @@
...
+using gen.send(), and the generator yields the transformed results to pass
+back to the caller.
+
+Transformations can also return auxiliary data using the `transform_with_aux`
+decorator. For example:
+
+@lu.transformation_with_aux
+def scale_transformer_aux(scale, x):
+  ans = yield (x / scale,)
+  yield ((x * scale,), "Auxiliary data: {}".format(x))
```
Comment (mattjj) [inline jax/linear_util.py]: Same here, I think you can return `x * scale, "Aux data"` though correct me if I'm wrong.

File: jax/linear_util.py (hunk 1 above)
Comment (hawkinsp) [reply jax/linear_util.py]: Done.

File: jax/linear_util.py (hunk 2 above)
Comment (hawkinsp) [reply jax/linear_util.py]: Done.

**Comment by hawkinsp** [comment]
(Dougal, feel free to send comments later.)

**Pull request merged**